python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -21,4 +21,212 @@
 
 """
 A class for holding restraints.
-"""
\ No newline at end of file
+"""
+
+from .._SireWrappers import Atom
+from ..Types import Length, Angle
+from .._SireWrappers import System as _System
+from ..Units.Length import nanometer
+from ..Units.Angle import degree, radian
+from ..Units.Energy import kj_per_mol
+
+class Restraint():
+    '''The Restraint class which holds the restraint information for the ABFE
+    calculations. Currently only Boresch type restraint is supported.
+
+    Boresch restraint is a set of harmonic restraints containing one bond, two
+    angle and three dihedrals, which comes from three atoms in the ligand
+    (l1, l2, l3) and three atoms in the protein (r1, r2, r3). The restraints
+    are arranged in the format of atom1-atom2 (equilibrium value, force constant):
+
+    Bonds: r1-l1 (r0, kr)
+    Angles: r2-r1-l1 (thetaA0, kthetaA), r1-l1-l2 (thetaB0, kthetaB)
+    Dihedrals: r3-r2-r1-l1 (phiA0, kphiA), r2-r1-l1-l2 (phiB0, kphiB), r1-l1-l2-l3 (phiC0, kphiC)
+
+    '''
+    def __init__(self, system, restraint_dict, type='Boresch'):
+        if not isinstance(system, _System):
+            raise TypeError("'system' must be of type 'BioSimSpace._SireWrappers.System'")
+        else:
+            # Store a copy of solvated system.
+            self._system = system.copy()
+
+        if type.lower() == 'boresch':
```

Comment:
Would it be better to use rest_type instead of type to be consistent with RestraintSearch, and to stay away from type()?